crates/sui-framework/packages/move-stdlib/sources/macros.move

```diff
@@ -84,7 +84,7 @@ public macro fun num_to_string($x: _): String {
     buffer.to_string()
 }
 
-public macro fun range_do($start: _, $stop: _, $f: |_|) {
+public macro fun range_do($start: _, $stop: _, $f: |_| -> _) {
```

Comment:
These changes feel fine to me, I just wonder if it is worth annotating, e.g. 
```suggestion
public macro fun range_do<R: drop>($start: _, $stop: _, $f: |_| -> R) {
```
Perhaps @tzakian or @cgswords will have opinions

Reply:
Wouldn't it force the return value to exist? Or `R: drop` can be `()`?

Reply:
It can be `()` for macros. Generics for macros have `$` for that reason to indicate they are a bit different, (so my suggestion should be)
```suggestion
public macro fun range_do<$R: drop>($start: _, $stop: _, $f: |_| -> $R) {
```

Reply:
Agreed this should have `drop` to help indicate the intended outcome to the programmer.